compiler-cli/src/compile_package.rs

```diff
@@ -56,12 +57,13 @@ pub fn command(options: CompilePackage) -> Result<()> {
 
 fn load_libraries(
     ids: &UniqueIdGenerator,
-    lib: &Path,
+    lib: &Utf8Path,
 ) -> Result<im::HashMap<SmolStr, ModuleInterface>> {
     tracing::info!("Reading precompiled module metadata files");
     let mut manifests = im::HashMap::new();
     for lib in fs::read_dir(lib)?.filter_map(Result::ok) {
-        let path = lib.path().join(paths::ARTEFACT_DIRECTORY_NAME);
+        let path = Utf8PathBuf::from_path_buf(lib.path().join(paths::ARTEFACT_DIRECTORY_NAME))
```

Comment:
fs::read_dir should return utf8 paths, there should be no `PathBuf` usage within our codebase if this library has been adopted.